unlock-app/src/components/interface/verification/invalidMembershipReason.ts

```diff
@@ -0,0 +1,37 @@
+import * as z from 'zod'
+import { isSignatureValidForAddress } from '~/utils/signatures'
+import { MembershipVerificationData } from '~/utils/verification'
+
+interface Options {
+  membershipData: any
+  rawData: string
+  data: z.infer<typeof MembershipVerificationData>
+  sig: string
+}
+
+export function invalidMembershipReason({
+  membershipData,
+  rawData,
```

Comment:
I don't think we need both `rawData` and `data`?

Reply:
We do. Both are used.